protoc-gen-gogoroach/main.go

```diff
@@ -0,0 +1,41 @@
+// This is a customized version of https://github.com/gogo/protobuf/blob/master/protoc-gen-gogoslick/main.go
+
+package main
+
+import (
+	"github.com/gogo/protobuf/vanity"
+	"github.com/gogo/protobuf/vanity/command"
+)
+
+func main() {
+	req := command.Read()
+	files := req.GetProtoFile()
+
+	vanity.ForEachFile(files, vanity.TurnOnMarshalerAll)
+	vanity.ForEachFile(files, vanity.TurnOnSizerAll)
+	vanity.ForEachFile(files, vanity.TurnOnUnmarshalerAll)
+
+	// either one of these breaks compilation of the descriptor proto :(
+	// vanity.ForEachFieldInFiles(files, vanity.TurnOffNullableForNativeTypesOnly)
+	// vanity.ForEachFile(files, vanity.TurnOffGoGettersAll)
+
+	// do we want to support unknown fields?
+	vanity.ForEachFile(files, vanity.TurnOffGoUnrecognizedAll)
+
+	// prefixes usually make sense
+	// vanity.ForEachFile(files, vanity.TurnOffGoEnumPrefixAll)
+
+	vanity.ForEachFile(files, vanity.TurnOffGoEnumStringerAll)
+	vanity.ForEachFile(files, vanity.TurnOnEnumStringerAll)
+
+	// generates broken code
+	// vanity.ForEachFile(files, vanity.TurnOnEqualAll)
+
+	// we have some custom stringers
```

Comment:
Sure, I'll try that.

There is some disagreement regarding the use of a custom binary in this way (see @petermattis' comments). Is there another way to use vanity?